src/python/pants/backend/go/util_rules/cgo.py

```diff
@@ -0,0 +1,858 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
```

Comment:
Worth transcribing Go unit tests for any of these intricate helpers and rules?

Reply:
There aren't actually any unit tests per se. Go does have a bunch of programs used for integration tests, but they are a bit hard to transcribe because they are very often written as a Go unit test which then invokes the Go toolchain internals on a project in a testdata directory. Something to do as cgo support matures.

https://github.com/pantsbuild/pants/issues/16829